File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s221078931_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct IntegralRect {
    int height, width;
    IntegralRect(int h, int w) : height(h), width(w) {}
    int diagonal() const {
        if (height >= width) return INT_MAX;
        return height * height + width * width;
    }
    bool operator<(const IntegralRect& o) const {
        int d1 = diagonal(), d2 = o.diagonal();
        if (d1 != d2) return d1 < d2;
        return height < o.height;
    }
};

int main() {
    vector<IntegralRect> rects(40000);
    for (int i = 0; i < 200; i++) {
        for (int j = 0; j < 200; j++) {
            int idx = i * 200 + j;
            rects[idx] = IntegralRect(i + 1, j + 1);
        }
    }
    sort(rects.begin(), rects.end());

    int h, w;
    while (cin >> h >> w) {
        if (h == 0 && w == 0) break;
        IntegralRect tmp(h, w);
        auto it = upper_bound(rects.begin(), rects.end(), tmp);
        if (it != rects.end()) {
            cout << it->height << " " << it->width << endl;
        }
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Python/s221078931_commented.py
import sys
import bisect

class IntegralRect:
    def __init__(self, h, w):
        self.height = h
        self.width = w

    def diagonal(self):
        if self.height >= self.width:
            return 2**31 - 1
        else:
            return self.height**2 + self.width**2

    def __lt__(self, other):
        d_self = self.diagonal()
        d_other = other.diagonal()
        if d_self != d_other:
            return d_self < d_other
        else:
            return self.height < other.height

def main():
    rects = []
    for h in range(1, 201):
        for w in range(1, 2嘲:
            rects.append(IntegralRect(h, w))
    rects.sort()

    for line in sys.stdin:
        h, w = map(int, line.split())
        if h == 0 and w == 0:
            break
        tmp = IntegralRect(h, w)
        idx = bisect.bisect_right(rects, tmp)
        if idx < len(rects):
            rect = rects[idx]
            print(rect.height, rect.width)

if __name__ == "__main__":
    main()
# End of Code
